### DISCUSSION TITLE
Correct encoding for binary choice data

### DISCUSSION THREAD

**Discussion by filippoferrari** [ilabcode/pyhgf#184] [Q&A] [answered]
Hi, 

I have a one armed-bandit task where, on each trial, an agent sees two stimuli A and B, chooses one of them and receives a reward with probability p(chosen) = 1-p(not chosen), i.e., at each trial one of the stimuli will lead to a reward. 

My data is stored as follows where `choice=0` shows outcomeA and `outcome=0/1` is nothing/win respectively, `win` being whether the chosen stimuli leads to a win

| trial | outcomeA | outcomeB | choice | win |
|-|-|-|-|-
| 1 | 0 | 1 | 1 | 1 |
| 2 | 1 | 0 | 1 | 0 |
| 3 | 1 | 0 | 0 | 1 |
| 4 | 1 | 0 | 1 | 0 |
| 5 | 1 | 0 | 0 | 1 |
|etc| | | | |

I am trying to estimate parameters using a binary hgf like as follows (based on the [multilevel tutorial](https://ilabcode.github.io/pyhgf/notebooks/3-Multilevel_HGF.html#a-multilevel-binary-hgf)) 
```python
hgf_logp_op = HGFDistribution(
    n_levels=2,
    model_type="binary",
    input_data=u,
    response_function=binary_softmax_inverse_temperature,
    response_function_inputs=responses,
)
```

I have tried various encoding:
- `input_data = outcomeA`, `response_function_inputs = choice`
- `input_data = win`, `response_function_inputs = choice`, this should be the same as the contingency space encoding described in the tapas HGF manual.pdf

but neither of these encodings seem to produce sensible parameters estimates for tonic_volatility_2 and inverse_temperature (the pymc model is defined similarly to the [parameter recovery tutorial](https://ilabcode.github.io/pyhgf/notebooks/4-Parameter_recovery.html#inference-from-the-simulated-behaviours)). 
In particular, I get all estimated values for the inverse_temperature to be 0<x<0.1, which makes me believe that the input data and response must be mismatched making the model behave randomly.  

What is the correct way to encode this type of data?

Thanks, 
Filippo

**Comment by LegrandNico** [comment]
Hi @filippoferrari , so we probably need to create a response function to handle this kind of design, The `HGFDistribution` class will only handle standard continuous or binary models. Do you have a dataset that you could share here (this can be simulated data) with a few participants, so I can make an example of multilevel modeling in the tutorial.

**Reply by filippoferrari** [reply, marked as answer]
Hi Nicolas, I found that by using outcomeB as your input_data everything works fine, as it correctly matches the contingency space encoding described in the Tapas manual.  
It already works for a single participants and for multilevel setup as in the tutorial. 

Thanks!
